analytics.js

```diff
@@ -355,31 +361,41 @@ class Analytics {
         object.clientIntegrationObjects,
       );
 
-      // send to all integrations now from the 'toBeProcessedByIntegrationArray' replay queue
-      for (let i = 0; i < succesfulLoadedIntersectClientSuppliedIntegrations.length; i += 1) {
-        try {
-          if (
-            !succesfulLoadedIntersectClientSuppliedIntegrations[i].isFailed ||
-            !succesfulLoadedIntersectClientSuppliedIntegrations[i].isFailed()
-          ) {
-            if (succesfulLoadedIntersectClientSuppliedIntegrations[i][methodName]) {
-              const sendEvent = !object.IsEventBlackListed(
-                event[0].message.event,
-                succesfulLoadedIntersectClientSuppliedIntegrations[i].name,
-              );
-
-              // Block the event if it is blacklisted for the device-mode destination
-              if (sendEvent) {
-                const clonedBufferEvent = cloneDeep(event);
-                succesfulLoadedIntersectClientSuppliedIntegrations[i][methodName](
-                  ...clonedBufferEvent,
-                );
-              }
-            }
-          }
-        } catch (error) {
-          handleError(error);
+      const intgWithoutTransformation = [];
+      const intgWithTransformation = [];
+
+      // Depending on transformation is connected or not
+      // create two sets of destinations
+      succesfulLoadedIntersectClientSuppliedIntegrations.forEach((intg) => {
```

Comment:
We should filter destinations based on the deny list before this step. There is no point in transforming events if they are to be blocked.

Reply:
Addressed